core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+                self.base_model.rename(self.original_model.name.replace('_BAG', ''))
+                X_val = X[~X.index.isin(X_train.index)]
+                y_val = y.loc[X_val.index]
+                if self.model_fit_time is not None:
+                    self.model_fit_time = self.model_fit_time / original_k_fold
+                if self.model_predict_time is not None:
+                    self.model_predict_time = self.model_predict_time / original_k_fold
+        else:
+            X_train, y_train = X, y
+
+        if subsample_val:
+            logger.log(30, f"\tNumber of validation samples {len(X_val)} is greater than {max_fi_samples}. Using {max_fi_samples} samples as validation data.")
+            if kwargs.get('num_classes', None) is not None:
+                # stratified sample if classification
+                combined = pd.concat([X_val, y_val], axis=1)
+                y_name = combined.columns[-1]
+                combined = combined.groupby(y_name).apply(lambda x: x.sample(frac=max_fi_samples/len(X_val), random_state=random_state)).reset_index(drop=True)
+                X_val, y_val = combined.drop(columns=[y_name]), combined[y_name]
+            else:
+                X_val = X_val.sample(n=max_fi_samples, random_state=random_state)
+                y_val = y_val.loc[X_val.index]
+        return X_train, y_train, X_val, y_val, subsample_train or subsample_val, kwargs
+
+    def setup(self, X: pd.DataFrame, y: pd.DataFrame, X_val: pd.DataFrame, y_val: pd.DataFrame, n_train_subsample: int, prune_threshold: float,
+              kwargs: dict) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame, bool, bool, dict]:
+        """
+        Modify training data, validation data, and model fit kwargs appropriately by subsampling, adding noise columns, replacing bagged
+        models, and more.
+        """
+        X, y, X_val, y_val, subsampled, kwargs = self.consider_subsampling(X=X, y=y, X_val=X_val, y_val=y_val, n_train_subsample=n_train_subsample, **kwargs)
+        auto_threshold = prune_threshold is None
+        if auto_threshold:
+            kwargs['feature_metadata'] = deepcopy(kwargs['feature_metadata']) if 'feature_metadata' in kwargs else None
+            X = add_noise_column(X, self.noise_prefix, self.rng, feature_metadata=kwargs.get('feature_metadata', None))
+            X_val = add_noise_column(X_val, self.noise_prefix, self.rng, feature_metadata=kwargs.get('feature_metadata', None))
+        X_fi, y_fi = (X, y) if self.is_bagged else (X_val, y_val)
+        if self.is_bagged:
+            # HACK: Consider not modifying params field
+            self.base_model.params['use_child_oof'] = False
+            self.base_model.params['save_bag_folds'] = True
+        if self.is_bagged and self.original_model._child_oof and self.model_fit_time is not None:
+            self.model_fit_time = self.model_fit_time * kwargs['k_fold']
```

Comment:
Why? The fit time of a bagged model is already the combined fit times of its children.

Reply:
This is because it only triggers when original model uses child_oof and we disable that for the proxy model.

Reply:
ok, add a comment here otherwise requires deep understanding of AG to understand.